aio_kavenegar/client.py

```diff
@@ -0,0 +1,182 @@
+            }
+        :param dict proxies: Dictionary mapping protocol to the URL of the proxy:
+            {
+                'http': 'http://192.168.1.10:3128',
+                'https': 'http://192.168.1.10:3129',
+            }
+        """
+        self.version: str = "v1"
+        self.host: str = host
+        self.apikey: str = apikey
+        self.apikey_mask: str = f"{apikey[:2]}********{apikey[-2:]}"
+        self.timeout: int = timeout
+        self.headers: dict = headers
+        if not proxies:
+            self.mounts: dict = None
+        else:
+            self.mounts: dict = {}
+            if http := proxies.get("http"):
+                self.mounts.update({"http://": httpx.HTTPTransport(proxy=http)})
+            if https := proxies.get("https"):
+                self.mounts.update({"https://": httpx.HTTPTransport(proxy=https)})
+        self.secure = secure
+
+    def __repr__(self) -> str:
+        return "kavenegar.AIOKavenegarAPI({!r})".format(self.apikey_mask)
+
+    def __str__(self) -> str:
+        return "kavenegar.AIOKavenegarAPI({!s})".format(self.apikey_mask)
+
+    def _pars_params_to_json(self, params: dict) -> dict:
+        """
+        Kavenegar bug, the api server expects the parameters in a JSON-like array format,
+        but the requests library form-encode each key-value pair
+
+        Params (dict):
+        { sender: ["30002626", "30002627", "30002727", ], }
+
+        request behavior:
+        sender=30002626&sender=30002627&sender=30002727
+
+        Server expectation:
+        sender=["30002626","30002627","30002727"]
+        """
+        # Convert lists to JSON-like strings
+        formatted_params = {}
+        for key, value in params.items():
+            if isinstance(value, (dict, list, tuple)):
+                formatted_params[key] = json.dumps(value)
+            else:
+                formatted_params[key] = value
+        return formatted_params
+
+    async def _request(
+        self,
+        action: Literal["sms", "verify", "call", "account"],
+        method: str,
+        params: dict = {},
+    ) -> dict:
+        params: dict = self._pars_params_to_json(params)
+        url = f"{"https://" if self.secure else "http://"}{self.host}/{self.version}/{self.apikey}/{action}/{method}.json"
+
+        try:
+            async with httpx.AsyncClient(mounts=self.mounts) as client:
+                http_response = await client.post(
+                    url,
+                    headers=self.headers,
+                    data=params,
+                    timeout=self.timeout,
+                )
+
+                try:
+                    response: KavenegarResponse = http_response.json()
+
+                    if response["return"]["status"] == 200:
+                        return response["entries"]
+                    else:
+                        raise APIException(
+                            f"APIException[{response["return"]["status"]}] {response["return"]["message"]}"
```

Comment:
Would you like to format the exception in the APIException class itself? Might be a better idea since it's not the client's job.

Reply:
@Alirezaja1384 , It's from the current code, I didn't write this at all, just reformatted by ruff in my machine.